### PR TITLE
feat: Notifications for repeatable program stages [DHIS2-10619]

### PR DESCRIPTION
**Pull request by zubaira**
PR adds a flag sendRepeatable in the notification template to achieve the desired functionality of repeatable notifications.
By default, it will be set to false.

### CODE REVIEW

**Review by enricocolasante — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/notification/ProgramNotificationTemplate.java [hunk 1]
```diff
@@ -68,6 +68,8 @@
 
     private Boolean notifyParentOrganisationUnitOnly;
 
+    private boolean sendRepeatable;
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/notification/ProgramNotificationTemplate.java:71]: Don't we need a Flyway migration here as `ProgramNotificationTemplate` is a persisted class? For primitive boolean we need to first add the field, populate with false values where null, then set to not-null.

Comment (zubaira) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/notification/ProgramNotificationTemplate.java:71]: yes, @larshelge flyway is needed.
We need to add the field and set it to false where null. By default, it's false for new templates.

**Review by gnespolino — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/program/hibernate/ProgramNotificationTemplate.hbm.xml [hunk 2]
```diff
@@ -36,6 +36,8 @@
 
     <property name="notifyUsersInHierarchyOnly" column="notifyusersinhierarchyonly" />
 
+    <property name="sendRepeatable" column="sendrepeatable" />
```
Comment (ameenhere) [inline dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/program/hibernate/ProgramNotificationTemplate.hbm.xml:39]: not-null="true" if its actually a non nullale column

**Review by ameenhere — CHANGES_REQUESTED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/program/hibernate/ProgramNotificationTemplate.hbm.xml (hunk 2 above)
Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/program/hibernate/ProgramNotificationTemplate.hbm.xml]: resolved

**Review by ameenhere — APPROVED** [review]

**Review by gnespolino — APPROVED** [review]

**Pull request merged**